### ISSUE TITLE
`install_runner_prerequisites` doesn't ensure chrony is installed

### ISSUE THREAD

**Issue by gjcolombo** [oxidecomputer/omicron#4094]
Reproduced by trying to run a dev cluster for the first time since #3825 landed (wow has it been a while!).

The cluster didn't come up on first boot; sled agent logs suggested that the issue was that `chronyc` wasn't on the path. Uninstalling and running `pkg install chrony` was enough to get me past that failure mode, so I suspect we need to add this to the list of packages in `install_runner_prerequisites.sh`.

This was *not* enough to get me up and running; after this `chronyc` ran in the NTP zone but consistently failed with `506 Cannot talk to daemon` errors. I didn't have chronyd running anywhere (in the GZ or otherwise) when I first tried this, so I uninstalled, tried to start the daemon, and reinstalled, but that didn't seem to fix it, so I ultimately decided to get unblocked by setting `skip_timesync` back to `true`. This needs more investigation and possibly a separate issue.

**Comment by citrus-it** [comment]
It's probable that you needed to restart `svc:/system/omicron/baseline:default` to get the updated `passwd` file (with the chrony user/group) into the archive used to seed zones. If that's it, then a synchronous restart of this service should also be done after installing chrony (and skipped if chrony was already installed, presumably).

**Comment by jordanhendricks** [comment]
I saw some similar errors while trying to deploy omicron (non-gimlet config, `skip_timesync = true`:


The ntp service inside the ntp zone was in maintenance:
```
root@oxz_ntp_4b3b1ff2-ab66-484f-a743-dd0d45af0da0:~# cat $(svcs -L ntp)                                                                                                     
[ Sep 25 02:26:42 Disabled. ]                                                                                                                                               
[ Sep 25 02:26:42 Rereading configuration. ]                                                                                                                                
[ Sep 25 02:26:44 Rereading configuration. ]                                          
[ Sep 25 02:26:49 Rereading configuration. ]                                                                                                                                
[ Sep 25 02:26:49 Enabled. ]                                                          
[ Sep 25 02:26:49 Executing start method ("/var/svc/method/svc-site-ntp start"). ]                                                                                          
NTP Service Configuration                                                             
-------------------------                                                             
 Servers: 0.pool.ntp.org                                                              
   Allow: fd00:1122:3344:100::/56                                                     
Boundary: true                                                                        
Template: /etc/inet/chrony.conf.boundary                                              
  Config: /etc/inet/chrony.conf                                                                                                                                             
* Updating logadm                                                                     
* Starting daemon                                                                                                                                                           
[ Sep 25 02:26:49 Method "start" exited with status 0. ]                                                                                                                    
2023-09-25T02:26:49Z chronyd version 4.3 starting (+CMDMON +NTP +REFCLOCK -RTC +PRIVDROP -SCFILTER +SIGND +ASYNCDNS -NTS +SECHASH +IPV6 -DEBUG)
2023-09-25T02:26:49Z Fatal error : Could not get user/group ID of chrony                                                                                                    
[ Sep 25 02:26:49 Stopping because all processes in service exited. ]                                                                                                       
[ Sep 25 02:26:49 Executing stop method ("/var/svc/method/svc-site-ntp stop 1857"). ]                                                                                       
NTP Service Configuration                                                             
-------------------------                                                             
 Servers: 0.pool.ntp.org                                                                                                                                                    
   Allow: fd00:1122:3344:100::/56                                                     
Boundary: true                                                                                                                                                              
Template: /etc/inet/chrony.conf.boundary                                              
  Config: /etc/inet/chrony.conf                                                       
* Stopping daemon                                                                     
[ Sep 25 02:26:49 Method "stop" exited with status 0. ]                                                                                                                     
[ Sep 25 02:26:49 Executing start method ("/var/svc/method/svc-site-ntp start"). ]                                                                                          
NTP Service Configuration                                                                                                                                                   
-------------------------                                                             
 Servers: 0.pool.ntp.org                                                              
   Allow: fd00:1122:3344:100::/56                                                     
Boundary: true                                                                        
Template: /etc/inet/chrony.conf.boundary                                              
  Config: /etc/inet/chrony.conf            
* Starting daemon                          
2023-09-25T02:26:49Z chronyd version 4.3 starting (+CMDMON +NTP +REFCLOCK -RTC +PRIVDROP -SCFILTER +SIGND +ASYNCDNS -NTS +SECHASH +IPV6 -DEBUG)
2023-09-25T02:26:49Z Fatal error : Could not get user/group ID of chrony                                                                                                    
[ Sep 25 02:26:49 Method "start" exited with status 0. ]                              
[ Sep 25 02:26:49 Stopping because all processes in service exited. ]                                                                                                       
[ Sep 25 02:26:49 Executing stop method ("/var/svc/method/svc-site-ntp stop 1860"). ]     
                                                            
```

**Comment by citrus-it** [comment]
> The ntp service inside the ntp zone was in maintenance:
...
> 2023-09-25T02:26:49Z Fatal error : Could not get user/group ID of chrony                                                                                                    

Restarting `svc:/system/omicron/baseline:default` is what should sort this after you have installed the chrony package - it updates the baseline archive used to populate new zones which will update the bundled passwd/group files.

**Comment by jordanhendricks** [comment]
Sorry I was not clear -- that output above was after restarting the baseline service. I'm going to take another lap today and see if I can get omicron stood up.
